models/mamba.py

```diff
@@ -0,0 +1,142 @@
+import torch
+import torch.nn.functional as F
+import lightning as L
+import wandb
+import einops
+import torchvision
+
+from mamba_ssm.models.mixer_seq_simple import MambaLMHeadModel
+from mamba_ssm.utils.generation import decode
+from constants import LOG2, VOCAB_SIZE
```

Comment:
Let's not make VOCAB_SIZE a global constant? We want the library to be more general and potentially work on grayscale datasets.

Reply:
there are no density estimation results for grayscale though right? does imagenet also 256 values for each channel